### ISSUE TITLE
[CLI] 'si seq deploy' doesn't work

### ISSUE THREAD

**Issue by a-tylenda** [scramjetorg/transform-hub#436]
**Describe the bug:**
When executing command `si seq deploy packages/reference-apps/endless-names-output`  I get error:

```
14:30 $ ts-node packages/cli/src/bin/index.ts seq deploy packages/reference-apps/endless-names-output
Request ok: http://127.0.0.1:8000/api/v1/sequence status: 200 OK
SequenceClient {
  _id: '2dafa1ac-10c0-4617-8b18-b361c23efaa2',
  host: HostClient {
    apiBase: 'http://127.0.0.1:8000/api/v1',
    client: ClientUtils {
      apiBase: 'http://127.0.0.1:8000/api/v1',
      fetch: [Function (anonymous)],
      normalizeUrlFn: [Function: normalizeUrl],
      log: [Object]
    }
  },
  sequenceURL: 'sequence/2dafa1ac-10c0-4617-8b18-b361c23efaa2'
}
Request ok: http://127.0.0.1:8000/api/v1/sequence/2dafa1ac-10c0-4617-8b18-b361c23efaa2/start status: 400 Bad Request
undefined
```

**Steps to Reproduce**
1. checkout to `devel` branch
2. start sth `scramjet-transform-hub`
3. `ts-node packages/cli/src/bin/index.ts seq deploy packages/reference-apps/endless-names-output`

**Reproduction repo/branch**
1. `devel`

**Expected behavior**
After executing `si seq deploy sequencePath` program should:
- pack Sequence if it is not packed into tar.gz (`si seq pack path/to/sequence`)
- send packed Sequence to STH (`si seq send -`)
- start Sequence (`si seq start -`)

**Version:**
 - STH version: 0.20.0

**Logs from STH**
```
2022-04-13T14:43:51.802Z INFO  Host New sequence incoming 
2022-04-13T14:43:51.805Z DEBUG Host Using DockerSequenceAdapter as sequence adapter 
2022-04-13T14:43:51.805Z TRACE DockerSequenceAdapter Initializing 
2022-04-13T14:43:51.805Z TRACE DockerodeDockerHelper Checking image [ 'scramjetorg/pre-runner:0.20.0' ]
2022-04-13T14:43:52.934Z TRACE DockerSequenceAdapter Initiazation done 
2022-04-13T14:43:52.941Z INFO  DockerSequenceAdapter Volume created [ '3c4d1bab-671e-4411-a10b-d924ba5e3569' ]
2022-04-13T14:43:52.941Z DEBUG DockerSequenceAdapter Starting PreRunner [
  {
    logLevel: 'trace',
    logColors: true,
    cpmUrl: '',
    cpmId: '',
    docker: { prerunner: [Object], runner: [Object], runnerImages: [Object] },
    identifyExisting: false,
    host: {
      hostname: '0.0.0.0',
      port: 8000,
      apiBase: '/api/v1',
      instancesServerPort: 8001,
      infoFilePath: '/tmp/sth-id.json'
    },
    instanceRequirements: { freeMem: 256, cpuLoad: 10, freeSpace: 128 },
    safeOperationLimit: 512,
    instanceAdapterExitDelay: 9000,
    runtimeAdapter: 'docker',
    sequencesRoot: '/home/atylenda/.scramjet_sequences',
    kubernetes: {
      namespace: 'default',
      authConfigPath: undefined,
      sthPodHost: undefined,
      runnerImages: [Object],
      sequencesRoot: '/home/atylenda/.scramjet_k8s_sequences',
      timeout: '0'
    }
  }
]
2022-04-13T14:43:54.842Z DEBUG DockerSequenceAdapter PreRunner response [
  {
    name: '@scramjet/endless-names-output',
    author: 'Scramjet <open-source@scramjet.org>',
    version: '0.20.0',
    keywords: null,
    description: '',
    main: 'index.js',
    engines: null,
    scramjet: { image: 'node' }
  }
]
2022-04-13T14:43:54.843Z TRACE DockerodeDockerHelper Checking image [ 'scramjetorg/runner:0.20.0' ]
2022-04-13T14:43:56.052Z INFO  Host Sequence identified [
  {
    type: 'docker',
    container: {
      image: 'scramjetorg/runner:0.20.0',
      maxMem: 512,
      exposePortsRange: [Array],
      hostIp: '0.0.0.0'
    },
    name: '@scramjet/endless-names-output',
    version: '0.20.0',
    engines: {},
    config: {},
    entrypointPath: 'index.js',
    id: '3c4d1bab-671e-4411-a10b-d924ba5e3569'
  }
]
2022-04-13T14:43:56.053Z DEBUG Host Request [
  'date: 2022-04-13T14:43:51.805Z, method: POST, url: /api/v1/sequence, status: 200'
]
2022-04-13T14:43:56.069Z INFO  Host Start sequence [
  '3c4d1bab-671e-4411-a10b-d924ba5e3569',
  '@scramjet/endless-names-output'
]
2022-04-13T14:43:56.071Z TRACE Host CSIController created [ '627e3ddb-1d96-44e9-a107-49787fc2ad63' ]
2022-04-13T14:43:56.070Z TRACE CommunicationHandler CommunicationHandler created 
2022-04-13T14:43:56.071Z DEBUG CSIController Constructor executed 
2022-04-13T14:43:56.073Z TRACE CSIController Streams hooked and routed 
2022-04-13T14:43:56.073Z TRACE CSIController Sequence initialized 
2022-04-13T14:43:56.073Z INFO  DockerInstanceAdapter Instance preparation done 
2022-04-13T14:43:56.075Z DEBUG DockerInstanceAdapter Runner will connect to STH on host OS using gateway [ '172.18.0.1' ]
2022-04-13T14:43:56.075Z DEBUG DockerInstanceAdapter Runner will start with envs [
  [
    'SEQUENCE_PATH=/package/index.js',
    'DEVELOPMENT=',
    'PRODUCTION=',
    'INSTANCES_SERVER_PORT=8001',
    'INSTANCES_SERVER_HOST=172.18.0.1',
    'INSTANCE_ID=627e3ddb-1d96-44e9-a107-49787fc2ad63'
  ]
]
2022-04-13T14:43:56.933Z TRACE DockerInstanceAdapter Container is running [ 'aad470134b3312280883da97cd2bdf094d8a3693045523ffc72b6a90402dba07' ]
2022-04-13T14:43:57.011Z ERROR DockerInstanceAdapter Docker container error:  [
  'Incorrect run argument: sequence path (/package/index.js) does not exists. \n'
]
2022-04-13T14:43:57.288Z DEBUG DockerInstanceAdapter Container exited [ 21 ]
2022-04-13T14:43:57.288Z DEBUG DockerInstanceAdapter Container returned non-zero status code [ 21 ]
2022-04-13T14:43:57.288Z ERROR CSIController Sequence finished with error [ 21 ]
2022-04-13T14:43:57.289Z ERROR Host CSIController errored [
  'Sequence entrypoint path index.js is invalid. Check "main" field in sequence package.json'
]
2022-04-13T14:43:57.290Z DEBUG Host Request [
  'date: 2022-04-13T14:43:56.059Z, method: POST, url: /api/v1/sequence/3c4d1bab-671e-4411-a10b-d924ba5e3569/start, status: 400'
]
```

**Comment by pietrzakacper** [comment]
`si deploy` command seem to work properly. The problem is with the directory that it is pointed to. In the example from the issue: `packages/reference-apps/endless-names-output` points to the source code, not to the built sources that it can pack.
The real workflow for deploying a package from source code would be something like this:
1. Transpile TS->JS for example to a `dist/` dir
2.. Copy package.json to `dist/`
3. In dist: `npm install --production`
4. `si seq deploy dist`

However to include such functionality in the deploy would require making assumptions about:
* project structure (e.g. is it possible to get all the required source code transpiled to dist, just by executing build command)
* dependency locking (whether someone uses yarn.lock or package-lock.json)
* ... and problem some more things that would come up with user problems

If we have a boilerplate for writing a sequence it could include `build` and `postbuild` scripts that would do steps from 1 to 3.
However on the CLI level we cannot make such assumptions (at least not without considering the before mentioned factors).
So it's reasonable to pass the path to the `dist` directory with all the source code, package.json, and dependencies already there.

***TLDR*** a fix for this issue is to run the command pointing to a build directory: `si seq deploy ./dist/reference-apps/endless-names-output`
